build.gradle

```diff
@@ -672,6 +672,7 @@ project(':streams') {
   archivesBaseName = "kafka-streams"
 
   dependencies {
+    compile project(':core')
```

Comment:
A dependency on `core` is definitely not OK (`core` actually means `broker and old clients` for historical reasons). However, it would be good to know what @guozhangwang thinks is the best solution.